File: odemapster-r2rml-rdb/src/main/java/es/upm/fi/dia/oeg/obdi/wrapper/r2rml/rdb/model/R2RMLTable.java
package es.upm.fi.dia.oeg.obdi.wrapper.r2rml.rdb.model;

import es.upm.fi.dia.oeg.obdi.core.sql.SQLFromItem.LogicalTableType;

public class R2RMLTable extends R2RMLLogicalTable {
	private String tableName;

	public R2RMLTable(String tableName, R2RMLTriplesMap owner) throws Exception {
		super(owner);
		if(tableName == null || tableName.equals("")) {
			throw new Exception("Empty table name specified!");
		}
		this.tableName = tableName;
		super.logicalTableType = LogicalTableType.TABLE_NAME;
	}

	@Override
	public String getValue() {
		return this.tableName;
	}
	
	
	
}

File: morph-r2rml/src/main/scala/es/upm/fi/dia/oeg/morph/r2rml/model/R2RMLTable.scala
package es.upm.fi.dia.oeg.morph.r2rml.model

import es.upm.fi.dia.oeg.morph.base.Constants
import es.upm.fi.dia.oeg.morph.r2rml.model.R2RMLLogicalTable


class R2RMLTable(tableName:String )
extends R2RMLLogicalTable(Constants.LogicalTableType.TABLE_NAME) {
		if(tableName == null || tableName.equals("")) {
			throw new Exception("Empty table name specified!");
		}
	override def getValue() : String = { this.tableName; }
}